content/docs/05.developer-guide/04.inputs.md

```diff
@@ -61,87 +63,142 @@ inputs:
 
 ## Input Properties
-These are the properties available for all input types:
 
-- `name`: The input name to be used with variables.
-- `required`: If the input is required. If required and no default value or no input is provided, the execution will not be created.
-- `defaults`: The default value if no input is provided. Must be a string.
-- `description`: A markdown description to document the input.
+Below is a list of properties available to all inputs regardless of their types:
+
+- `name`: the input parameter name — this property is important as it's used to reference the input variables in your flow, e.g. `{{ inputs.user }}` will reference the input parameter named `user`
+- `type`: the data type of the input parameter, as described in the last section
+- `required`: whether the input is required or optional; if required is set to true and no default value is configured and also no input is provided at runtime, the execution will not be created as Kestra cannot know what value to use
+- `defaults`: the default value that will be used if no custom input value is provided at runtime; this value must be provided as a string and will be coerced to the desired data type specified using the `type` property
+- `description`: a markdown description to document the input.
+
 
 ## Input validation
 
-In addition to the standard validation rules explained above, some input types can be configured with validation rules that will be enforced at execution time.
+Kestra validates the `type` of each input. In addition to the type validation, some input types can be configured with validation rules that will be enforced at execution time.
+
+- `STRING`: a `validator` property allows the addition of a validation [regex](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html).
+- `INT`: `min` and `max` properties allow the addition of minimum and maximum value ranges
+- `FLOAT`: `min` and `max` properties allow the addition of minimum and maximum ranges
+- `DURATION`: `min` and `max` properties allow the addition of minimum and maximum ranges
+- `DATE`: `after` and `before` properties help you ensure that the input value is within the allowed date range
+- `TIME`: `after` and `before` properties help you ensure that the input value is within the allowed date range
+- `DATETIME`: `after` and `before` properties help you ensure that the input value is within the allowed date range.
+
+## Example: using input validators in your flows
+
+To ensure that your input value is within a certain `integer` value range, you can use the `min` and `max` properties. Similarly, to ensure that your string input matches a Regex pattern, you can provide a custom regex `validator`. The following flow demonstrates how this can be accomplished:
```

Comment:
```suggestion
To ensure that your input value is within a certain `integer` value range, you can use the `min` and `max` properties. Similarly, to ensure that your string input matches a regex pattern, you can provide a custom regex `validator`. The following flow demonstrates how this can be accomplished:
```